src/CholeskySpace.jl

```diff
@@ -0,0 +1,97 @@
+using LinearAlgebra: diagm, diag, eigen, eigvals, eigvecs, Symmetric, Diagonal, factorize, tr, norm, cholesky, LowerTriangular
+
+@doc doc"""
+    CholeskySpace{N} <: Manifold
+
+the manifold of lower triangular matrices with positive diagonal and
+a metric based on the cholesky decomposition. The formulae for this manifold
+are for example summarized in Table 1 of
+> Lin, Zenhua: Riemannian Geometry of Symmetric Positive Definite Matrices via
+> Cholesky Decomposition, arXiv: 1908.09326
+"""
+struct CholeskySpace{N} <: Manifold end
+CholeskySpace(n::Int) = CholeskySpace{N}()
+
+@generated representation_size(::CholeskySpace{N}) where N = (N,N)
+
+@generated manifold_dimension(::CholeskySpace{N}) where N = N*(N+1)/2
+
+@doc doc"""
+    inner(M,x,v,w)
+
+computes the inner product on the [`CholeskySpace`](@ref) `M` at the
+lower triangular matric with positive diagonal `x` and the two tangent vectors
+`v`,`w`, i.e they are both lower triangular matrices with arbitrary diagonal.
+The formula reads
+
+````math
+    g_{x}(v,w) = \sum_{i>j} v_{ij}w_{ij} + \sum_{j=1}^m v_{ii}w_{ii}x_{ii}^{-2}
+````
+"""
+inner(::CholeskySpace{N},x,v,w) where N = sum(LowerTriangular(v).*LowerTriangular(w)) + sum(diag(v).*diag(w)./( diag(x).^2 ))
+
+@doc doc"""
+    distance(M,x,y)
+
+computes the Riemannian distance on the [`CholeskySpace`](@ref) `M` between two
+matrices `x`, `y` that are lower triangular with positive diagonal. The formula
+reads
+
+````math
+d_{\mathcal M}(x,y) = \sqrt{
+\sum_{i>j} (x_{ij}-y_{ij})^2 + 
+\sum_{j=1}^m (\log x_{jj} - \log_{y_jj})^2
+}
+````
+"""
+distance(::CholeskySpace{N},x,y) where N = sqrt(
+  sum( (LowerTriangular(x) - LowerTriangular(y)).^2 ) + sum( (log.(diag(x)) - log(diag(y))).^2 )
+)
+
+@doc doc"""
+    exp!(M,y,x,v)
+
+compute the exponential map on the [`CholeskySpace`](@ref) `M` eminating from
+the lower triangular matrix with positive diagonal `x` towards the lower triangular
+matrx `v` and return the result in `y`. The formula reads
+
+````math
+\exp_x v = \lfloor x \rfloor + \lfloor v \rfloor
++\operatorname{diag}(x)\operatorname{diag}(x)\exp{ \operatorname{diag}(v)\operatorname{diag}(x)^{-1}}
+````
+where $\lfloor x\rfloor$ denotes the lower triangular matrix of $x$ and
+$\opertorname{diag}(x)$ the diagonal matrix of $x$
+"""
+function exp!(::CholeskySpace{N},y,x,v) where N
+    y .= LowerTriangular(x) + LowerTriangular(v) + diagm(x)*diagm(exp.(diag(v)./diag(x)))
```

Comment:
`Diagonal(x)` makes a diagonal matrix without actually allocating the whole matrix (in costrast with `diagm(x)`).

Reply:
`log!` has a similar issue